### DISCUSSION TITLE
User type in Graphql Query

### DISCUSSION THREAD

**Discussion by utkarsha-ubale-sp** [community/community#72939] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am trying to get list of the members of organisation and outside collaborators using below two queries : 

List Outside Collaborators : 
query ListOutsideCollabortors($org: String!, $cursor: String) {
        organization(login: $org) {
            url
            login
            repositories(first: 100, after: $cursor) {
                pageInfo {
                    endCursor
                    hasNextPage
                }
                nodes {
                    name
                    collaborators(affiliation: OUTSIDE, first: 100) {
                        edges {
                            permission
                        }
                        nodes {
                            databaseId
							login
							name
							id
							email
							twitterUsername
							url
							bio
							bioHTML
							location
							company
							companyHTML
							isBountyHunter
							isHireable
							isEmployee
							isSiteAdmin
							isGitHubStar
							isViewer
							isCampusExpert
							isDeveloperProgramMember
							hasSponsorsListing
							followers {
								totalCount
							}
							following {
								totalCount
							}
							gists {
								totalCount
							}
							repositories {
								totalCount
							}
							watching {
								totalCount
							}
							estimatedNextSponsorsPayoutInCents
							monthlyEstimatedSponsorsIncomeInCents
							websiteUrl
							createdAt
							updatedAt
                        }
                    }
                }
            }
        }
    }

List Organisation members : 
query ListOrganizationMembers($org: String!, $cursor: String, $first: Int = 100) {
		viewer {
			organization(login: $org) {
				membersWithRole(after: $cursor, first: $first) {
					totalCount
					edges {
						hasTwoFactorEnabled
						role
						node {
							databaseId
							login
							name
							id
							email
							twitterUsername
							url
							bio
                            
							bioHTML
							location
							company
							companyHTML
							isBountyHunter
							isHireable
							isEmployee
							isSiteAdmin
							isGitHubStar
							isViewer
							isCampusExpert
							isDeveloperProgramMember
							hasSponsorsListing
							followers {
								totalCount
							}
							following {
								totalCount
							}
							gists {
								totalCount
							}
							repositories {
								totalCount
							}
							watching {
								totalCount
							}
							estimatedNextSponsorsPayoutInCents
							monthlyEstimatedSponsorsIncomeInCents
							websiteUrl
							createdAt
							updatedAt
						}
					}
					pageInfo {
						endCursor
						hasNextPage
                    }
                    
				}
			}
		}
	}

I want to know, is there any attribute which would help to determine if it is a member of org or outside collab?

**Comment by b8zeek** [comment]
By the **GitHub**'s GraphQL API documentation, the `affiliation` field in the `collaborators` connection determines whether a user is an outside collaborator or a member of the organization. So, the 1st query actually returns collaborators because you're querying it with **"OUTSIDE"** value while the 2nd query returns members because you're using `membersWithRole`. In 2nd query, members query, you can also use `affiliation` field with **"MEMBER"** in order to get the members.

Cheers.

🍀

**Reply by utkarsha-ubale-sp** [reply]
Thank you for your response. I was wondering if there is a common attribute in nodes, which would help in differentiating the two.

**Reply by b8zeek** [reply]
Yeah, that would be the `affiliation` attribute. 😄 If you have any questions, let me know, otherwise I would appreciate if you can close the discussion.

Cheers. 🍺

**Reply by utkarsha-ubale-sp** [reply]
Thanks!

**Reply by utkarsha-ubale-sp** [reply]
I have one more question here, to get the information of any particular user, I am using below query : 

query ReadOrganizationUser($org: String!, $nativeIdentifier: String!, $first: Int) {
		user(login: $nativeIdentifier) {
			databaseId
			login
			name
			id
			email
			twitterUsername
			url
			bio
			bioHTML
			location
			company
			companyHTML
			isBountyHunter
			isHireable
			isEmployee
			isSiteAdmin
			isGitHubStar
			isViewer
			isCampusExpert
			isDeveloperProgramMember
			hasSponsorsListing
			followers {
				totalCount
			}
			following {
				totalCount
			}
			gists {
				totalCount
			}
			repositories {
				totalCount
			}
			watching {
				totalCount
			}
			estimatedNextSponsorsPayoutInCents
			monthlyEstimatedSponsorsIncomeInCents
			websiteUrl
			createdAt
			updatedAt
		}
		organization(login: $org) {
			teams(userLogins: [$nativeIdentifier], first: $first, rootTeamsOnly: false) {
				totalCount
				nodes {
					databaseId
					slug
					name
					id
				}
				pageInfo {
					endCursor
					hasNextPage
				}
			}
		}
	}


I don't find any attribute to determine the type. "affiliation" is not an attribute here. Can you please let me know if there is any such attribute.
Thanks in advance!

**Comment by b8zeek** [comment, marked as answer]
I do not think that you can fetch both members and outside contributors in the same query. Below, you'll find the queries to fetch both groups.

```
query OrganizationMembers($org: String!) {
  organization(login: $org) {
    membersWithRole {
      nodes {
        user {
          login
        }
      }
    }
  }
}

query RepositoryContributors($org: String!, $repo: String!) {
  repository(owner: $org, name: $repo) {
    collaborators {
      nodes {
        login
      }
    }
  }
}
```

Hope this helps. Cheers. 🍺 

🍀

**Reply by ghostinhershell** [reply]
Cheers to you! Thank you for contributing here! 🚀

**Reply by b8zeek** [reply]
No problem ser, a huge fan of GitHub, very thankful and I plan to dedicate much more time to the community and open source projects! 🍀

**Reply by ghostinhershell** [reply]
Really excited to see all of the ways that you contribute to the community 🚀

**Reply by b8zeek** [reply]
Thanks, I appreciate the support. ☺️ In case you have an idea or you're willing to brainstorm, we can build something cool together. You can reach out to me, and perhaps we can talk.

🍀
